File: crypto/src/main/java/io/oversec/one/crypto/UserInteractionRequiredException.java
package io.oversec.one.crypto;

import android.app.PendingIntent;

import java.util.List;

public class UserInteractionRequiredException extends Exception {

    private final PendingIntent mPendingIntent;
    private Long[] mPublicKeyIds;

    public UserInteractionRequiredException(PendingIntent pi, List<Long> pkids) {
        mPendingIntent = pi;
        if (pkids != null) {
            mPublicKeyIds = new Long[pkids.size()];
            pkids.toArray(mPublicKeyIds);
        }
    }

    public UserInteractionRequiredException(PendingIntent pi, Long[] pkids) {
        mPendingIntent = pi;
        mPublicKeyIds = pkids;
    }


    public UserInteractionRequiredException(PendingIntent pi) {
        mPendingIntent = pi;
    }

    public PendingIntent getPendingIntent() {
        return mPendingIntent;
    }


}

File: crypto/src/main/java/io/oversec/one/crypto/UserInteractionRequiredException.kt
package io.oversec.one.crypto

import android.app.PendingIntent

open class UserInteractionRequiredException : Exception {

    val pendingIntent: PendingIntent
    private var mPublicKeyIds: LongArray? = null  //TODO: not longer used ?!

    constructor(pi: PendingIntent, pkids: List<Long>?) {
        pendingIntent = pi
        mPublicKeyIds = pkids?.toLongArray()
    }

    constructor(pi: PendingIntent, pkids: LongArray) {
        pendingIntent = pi
        mPublicKeyIds = pkids
    }


    constructor(pi: PendingIntent) {
        pendingIntent = pi
    }


}
